Jint/Native/Array/ArrayInstance.cs

```diff
@@ -120,14 +117,16 @@ public override bool DefineOwnProperty(string propertyName, PropertyDescriptor d
 
                 if (_array.Count < oldLen - newLen)
                 {
-                    var keys = _array.Keys.ToArray();
+                    var keys = keyCache.Value;
+                    keys.Clear();
+                    keys.AddRange(_array.Keys);
                     foreach (var key in keys)
                     {
                         uint keyIndex;
                         // is it the index of the array
                         if (IsArrayIndex(key, out keyIndex) && keyIndex >= newLen && keyIndex < oldLen)
```

Comment:
Is it possible that we don't need to call `IsArrayIndex` at all here? `keys` is a list of `uint`, so the end result will always be a valid `uint`.